lib/pages/new_group/new_group.dart

```diff
@@ -68,6 +78,30 @@ class NewGroupController extends State<NewGroup> with ComparablePresentationCont
     });
   }
 
+  void listenCreateRoom() {
+    createRoomStreamController.streamController.stream.listen((event) {
+      Logs().d('NewGroupController::createRoom() - event: $event');
+      event.fold(
+        (failure) => {
+          createRoomStreamController.dispose(),
+          Logs().e('NewGroupController::createRoom() - failure: $failure'),
+        },
+        (Success success) => {
+          Logs().d('NewGroupController::createRoom() - success: $success'),
+          if (success is CreateRoomSuccess) {
+            if (success.event == CreateRoomSuccessEventType.createRoomSuccess) {
```

Comment:
I am not a fan of `ton of nested if` clause. Please separate the `success state` (each state = each kind of message you want to deliver/handle in UI)

Reply:
I put a switch statement instead